x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/request/ElasticInferenceServiceRequest.java

```diff
@@ -36,13 +38,18 @@ public final HttpRequest createHttpRequest() {
 
         var productOrigin = metadata.productOrigin();
         var productUseCase = metadata.productUseCase();
+        var esVersion = metadata.esVersion();
 
         if (Objects.nonNull(productOrigin) && productOrigin.isEmpty() == false) {
-            request.setHeader(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER, metadata.productOrigin());
+            request.setHeader(Task.X_ELASTIC_PRODUCT_ORIGIN_HTTP_HEADER, productOrigin);
         }
 
         if (Objects.nonNull(productUseCase) && productUseCase.isEmpty() == false) {
-            request.addHeader(X_ELASTIC_PRODUCT_USE_CASE_HTTP_HEADER, metadata.productUseCase());
+            request.addHeader(X_ELASTIC_PRODUCT_USE_CASE_HTTP_HEADER, productUseCase);
+        }
+
+        if (Objects.nonNull(esVersion) && esVersion.isEmpty() == false) {
```

Comment:
super nit: There's a help we could use `Strings.isNullOrEmpty` for these if blocks